SpecialOperations2.Altis/sos/mission/functions/fn_taskCaptureCompound.sqf

```diff
@@ -0,0 +1,143 @@
+{
+    if(_unitCount > _maxUnits) exitWith {};
+    
+    _occupyBuilding = [0, 100] call BIS_fnc_randomInt;
+    if (_occupyBuilding <= _buildingCoverage) then {
+        _building = _x;
+        _class = typeOf _building;
+        _positions = getNumber (missionConfigFile >> "CfgBuildings" >> _configName >> _class >> "positions");
+        _specialUnits = getArray (missionConfigFile >> "CfgBuildings" >> _configName >> _class >> "specialUnits");
+        _specialUnitsPositions = getArray (missionConfigFile >> "CfgBuildings" >> _configName >> _class >> "specialUnitsPositions");
+        _spawnSpecialUnit = getNumber (missionConfigFile >> "CfgBuildings" >> _configName >> _class >> "spawnPercentSpecialUnit");
+    
+        _numberOfUnits = floor (_positions / _occupyNumber);
+        if(_numberOfUnits == 0) then {_numberOfUnits = _numberOfUnits + 1};
+        for [{_i = 0}, {_i < _numberOfUnits}, {_i = _i + 1}] do {
+            _ai = nil;
+            _randomPos = nil;
+        
+            _isSpecial = [0, 100] call BIS_fnc_randomInt;
+            if(_isSpecial <= _spawnSpecialUnit) then {
+                _ai = createGroup east createUnit [_specialUnits call BIS_fnc_selectRandom, [0.0, 0.0, 0.0], [], 0, "CAN_COLLIDE"];
+                _randomPos = _specialUnitsPositions call BIS_fnc_selectRandom;
+            } else {
+                _ai = createGroup east createUnit [_units call BIS_fnc_selectRandom, [0.0, 0.0, 0.0], [], 0, "CAN_COLLIDE"];
+                _randomPos = [0, (_positions - 1)] call BIS_fnc_randomInt;
+            };
+            _ai setPosATL (_building buildingPos _randomPos);
+            _ai setUnitPos "UP";
+            _ai setDir random 360.0;
+            
+            _unitCount = _unitCount + 1;
+        };
+    };
+} forEach _base;
+
+//Get middle position of the first and last building inside the base
+_beginPos = position (_base select 0);
+_endPos = position (_base select ((count _base) - 1));
+
+_centerX = floor (((floor(_beginPos select 0)) + (floor (_endPos select 0))) / 2);
+_centerY = floor (((floor(_beginPos select 1)) + (floor (_endPos select 1))) / 2);
+_centerZ = floor ((_beginPos select 2));
+_pos = [_centerX, _centerY, _centerZ];
+
+//Create zone and patrols
+_radius = _radius + 20;
+
+_zone = [_pos, [_radius, _radius]] call SOS_fnc_createZone;
+_groups = ["OIA_InfTeam_AT", "OIA_InfTeam", "OIA_InfTeam_AA"];
+for [{_i = 0}, {_i < _maxPatrols}, {_i = _i + 1}] do {
+    _position = [_pos, 0.0, 50.0, 2.0, 2.0] call SOS_fnc_findSafePosition;
+    _group = [_position, EAST, (configfile >> "CfgGroups" >> "East" >> "OPF_F" >> "Infantry" >> _groups call BIS_fnc_selectRandom)] call BIS_fnc_spawnGroup;
+    _group setVariable ["GAIA_ZONE_INTEND", [format ["%1", _zone], "NOFOLLOW"]];
+};
+
+//Create task, marker and trigger
+_task = ["CaptureCompound", _pos] call SOS_fnc_addTask;
+
+_marker = createMarker ["SOS_MISSION_ASSAULT_BASE", _pos];
```

Comment:
this is no longer needed!

in the next push you can define what map marker your task has in the **CfgTaskDescriptions**. We can work something out so it can support ellipse markers.